src/widgets/Tabs/VaccineSupplementalData.js

```diff
@@ -37,6 +43,7 @@ const VaccineSupplementalDataComponent = ({ onCancel, onComplete, siteSchema })
         style={{ flex: 1 }}
       >
         <JSONForm
+          formData={formData}
```

Comment:
If I edit the form then step back a step or two, then when I reenter this step, the state is the same, which I'm unsure is the desired behaviour.

The original `PatientEdit` is maybe good to follow?

The `JSONForm` is a bit of a nightmare.. it has internal state and doesn't cause re-renders of its parent, so by updated `formData` in the `Suppliermentary` parent component, we end up maintaining that state if we switched patient.

If I remember correctly, `PatientEdit` seeds the data by putting the most recent filled out form into the reducer when a patient is selected, then seeding `JSONForm` with that data. When editing the form, the form is updated in the reducer state. When the patient changes, the seed changes in the reducer, which updates the form correctly... at the moment the `useState` won't be re-seeded, as `VaccineSuppl...` is never re-mounted only re-rendered.

Hope that makes sense (and is accurate?!?!)

Reply:
Yeah, it probably makes sense to reset the form when the patient is changed (at least that's what I think I'd expect as a user). 

I didn't realise the PCD was supposed to be seeded based on previous data (but I can see the code that does it) - I think my PCD form (even when filled out) is resulting in 'undefined' formData 🙀  I'm not sure what's going on but I'll take a look 🤔

Reply:
Fixed the prepopulation to not persist between patients: 7dc0ad8 9f6f033

PCD problem was just me not setting up the 'PCD' PatientEvent in mSupply correctly (if it's not set up they don't get saved 🤣 )